c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -583,6 +697,15 @@ class DeviceCachingAllocator {
       process_events();
     }
 
+    // try to allocate according to plan
+    if (memory_planner.active) {
+      Block* plan_block = memory_planner.malloc(size, device, stream);
+      // nullptr indicates event is not allocated according to plan
+      if (plan_block) {
+        return plan_block;
+      }
```

Comment:
this is not planned, do we need to print a warning?

Reply:
Do we expect the allocations to be perfectly periodic? If not could it be annoying to keep printing warning?